modules/transactionpool/update.go

```diff
@@ -8,6 +8,124 @@ import (
 	"github.com/NebulousLabs/Sia/types"
 )
 
+// findSets takes a bunch of transactions (presumably from a block) and finds
+// all of the separate transaction sets within it. Set does not check for
+// conflicts.
+func findSets(ts []types.Transaction) [][]types.Transaction {
```

Comment:
I feel like this is a graph theory problem: "given a set of edges, partition the edges into a set of connected graphs". But I can't find any literature about it.